Add notes to CSV Export. Move save folder to Downloads. Perform additional checks before saving. Fix bug with mmol. Closes: #290.

PaoloRotolo opened the pull request:
Please review

emartynov commented on app/src/main/java/org/glucosio/android/presenter/ExportPresenter.java:
```diff
@@ -130,6 +138,21 @@ protected void onPostExecute(String filename) {
+        return file.exists() || file.mkdirs();
+    }
+
+    private boolean arePermissionsGranted() {
```
Not used method?

PaoloRotolo replied:
Remove, thanks.

emartynov commented on app/src/main/java/org/glucosio/android/tools/ReadingToCSV.java:
```diff
@@ -42,72 +42,84 @@ public ReadingToCSV(Context mContext) {
+                        osw.append(',');
 
-                        osw.append(readings.get(i).getReading() + "mg/dL");
+                        osw.append(dateTool.convertRawTime(reading.getCreated() + ""));
```
Printing date time two times?

PaoloRotolo replied:
The first one converts the date. The second one the time:
```
osw.append(dateTool.convertRawDate(reading.getCreated() + ""));
osw.append(dateTool.convertRawTime(reading.getCreated() + ""));
```

emartynov commented on app/src/main/java/org/glucosio/android/tools/ReadingToCSV.java:
```diff
@@ -42,72 +42,84 @@ public ReadingToCSV(Context mContext) {
                         osw.append(',');
 
-                        osw.append(dateTool.convertRawDate(readings.get(i).getCreated() + ""));
+                        osw.append("mg/dL");
```
Are we really taking it in mg/dL? Or we actually store it in different formats?

emartynov commented:
Just minor comments. And I don't understand the fix again.

PaoloRotolo commented:
So there were a typo in the CSV class when exporting with mmol as unit of measurement. The result was a broken CSV table. I fixed the typo and added new fields for Notes and other info. Also now the exported file goes in the Download folder (since on some devices we haven't permission to write outside).
